operator/helper/multiline.go

```diff
@@ -226,16 +212,53 @@ func SplitNone(maxLogSize int) bufio.SplitFunc {
 	}
 }
 
-// NewLineEndSplitFunc creates a bufio.SplitFunc that splits an incoming stream into
-// tokens that end with a match to the regex pattern provided
-func NewLineEndSplitFunc(re *regexp.Regexp, flushAtEOF bool, force *Flusher) bufio.SplitFunc {
+func flusherSplitFunc(force *Flusher, splitFunc bufio.SplitFunc) bufio.SplitFunc {
```

Comment:
Should be a receiver on the Flusher, instead of passing it in?

Reply:
I moved it and refactored `trimWhitespaces` and `getSplitFunc` to simplify usage of new function